e2e/test/helpers/RetryOperationHelper.cs

```diff
@@ -15,14 +16,22 @@ namespace Microsoft.Azure.Devices.E2ETests.Helpers
     public class RetryOperationHelper
     {
         /// <summary>
-        /// Rety an async operation based on the retry strategy supplied.
+        /// Retry an async operation based on the retry strategy supplied.
         /// </summary>
+        /// <remarks>
+        /// This is for E2E tests of provisioning service clients .
+        /// </remarks>
         /// <param name="asyncOperation">The async operation to be retried.</param>
         /// <param name="retryPolicy">The retry policy to be applied.</param>
         /// <param name="retryableExceptions">The exceptions to be retried on.</param>
         /// <param name="logger">The <see cref="MsTestLogger"/> instance to be used.</param>
-        /// <returns></returns>
-        public static async Task RetryOperationsAsync(Func<Task> asyncOperation, IRetryPolicy retryPolicy, HashSet<Type> retryableExceptions, MsTestLogger logger)
+        /// <param name="cancellationToken">The cancellation token to cancel the operation.</param>
+        public static async Task RetryOperationsAsync(
+            Func<Task> asyncOperation,
+            IRetryPolicy retryPolicy,
+            HashSet<Type> retryableExceptions,
+            MsTestLogger logger,
+            CancellationToken cancellationToken = default)
```

Comment:
with this change, now it's able to stop retries if the user's cancellation token has been signaled.